pkg/3rdparty/toml/encode.go

```diff
@@ -379,16 +394,19 @@ func (enc *Encoder) eArrayOfTables(key Key, rv reflect.Value) {
 }
 
 func (enc *Encoder) eTable(key Key, rv reflect.Value) {
+	enc.inTable = true
```

Comment:
Use a tmp variable to cache the init state of `enc.inTable`

Reply:
Once a table appears, no other types are allowed to appear even after a nested table. So it is unnecessary， e.g.
```toml
[a]
  [a.b]
    a = 1
  a = 1 # child of table `a` is not allowed
```